website/static/js/institutionSignIn.js

```diff
@@ -18,11 +18,26 @@ var ViewModel = function() {
                 isCors: true
             }
         ).done(function (response) {
-            self.instNames(response.data.map(function(item){
-                var name = item.attributes.name;
-                self.insts[name] = item.attributes.auth_url + '&target=' + encodeURIComponent(window.contextVars.institution_redirect);
-                return name;
-            }));
+            self.instNames(
+                response.data.map(function(item){
+                    var name = item.attributes.name;
+                    self.insts[name] = item.attributes.auth_url + '&target=' + encodeURIComponent(window.contextVars.institution_redirect);
+                    self.insts[item.id] = name;
+                    if (item.attributes.auth_url){
+                        return name;
+                    }
+                }).filter(function (item) {
+                   return typeof item !== 'undefined';
```

Comment:
Oh, I see now; it's possible that the `map` function will return `undefined`. How about doing the filter first?

```js
response.data.filter(function(item) { return item.attributes.auth_url;}).map(function(item) {
    // ...
})
```